optuna/samplers/tpe/sampler.py

```diff
@@ -218,15 +218,10 @@ def _sample_discrete_uniform(self, distribution, below, above):
     def _sample_int(self, distribution, below, above):
         # type: (distributions.IntUniformDistribution, np.ndarray, np.ndarray) -> int
 
-        low = 0 - 0.5 
-        high = int((distribution.high - distribution.low) / distribution.step) + 0.5
-
-        above = (above - distribution.low) / distribution.step
-        below = (below - distribution.low) / distribution.step
-        best_sample = (
-            self._sample_numerical(low, high, below, above) * distribution.step + distribution.low
```

Comment:
May I have the reason why did you revert this change?

Reply:
This update is based on https://github.com/optuna/optuna/pull/910#discussion_r396975506

Then, the current implementation is following 
https://github.com/optuna/optuna/pull/910#discussion_r396977692

Reply:
I see. I misunderstood the change. LGTM 👍